segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -323,20 +323,19 @@ public void updateSegmentAttribute(UpdateSegmentAttribute updateSegmentAttribute
         long trace = LoggerHelpers.traceEnter(log, operation, updateSegmentAttribute);
         val update = new AttributeUpdate(attributeId, AttributeUpdateType.ReplaceIfEquals, newValue, expectedValue);
         segmentStore.updateAttributes(segmentName, Collections.singletonList(update), TIMEOUT)
-                .whenComplete((v, e) -> {
-                    LoggerHelpers.traceLeave(log, operation, trace, e);
-                    if (e == null) {
+                    .thenAccept(v -> {
+                        LoggerHelpers.traceLeave(log, operation, trace);
                         connection.send(new SegmentAttributeUpdated(requestId, true));
-                    } else {
+                    })
+                    .exceptionally(e -> {
                         if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
                             log.debug("Updating segment attribute {} failed due to: {}", update, e.getMessage());
                             connection.send(new SegmentAttributeUpdated(requestId, false));
                         } else {
                             handleException(requestId, segmentName, operation, e);
                         }
-                    }
-                })
-                .exceptionally(e -> handleException(requestId, segmentName, operation, e));
+                        return null;
+                    });
```

Comment:
Previously if the connection.send threw an exception handleException would be invoked. This is no longer the case in the new version.

Reply:
updated the code to have a try catch within the `whenComplete` block.